provider-service/base/pkg/streams/sinks/error_sink.go

```diff
@@ -0,0 +1,30 @@
+package sinks
+
+import (
+	"context"
+	"github.com/sky-uk/kfp-operator/argo/common"
+)
+
+type ErrorSink struct {
+	context context.Context
+	in      chan error
+}
+
+func NewErrorSink(ctx context.Context, inChan chan error) *ErrorSink {
+	errorSink := &ErrorSink{context: ctx, in: inChan}
+
+	go errorSink.Log()
+
+	return errorSink
+}
+
+func (ls ErrorSink) In() chan<- error {
+	return ls.in
+}
+
+func (ls ErrorSink) Log() {
```

Comment:
```suggestion
func (es ErrorSink) In() chan<- error {
	return ls.in
}

func (es ErrorSink) Log() {
```

Reply:
changed in 847762891ac9e4c53e697be2fbd16d03bfb70260